lib/dsc-lib/src/functions/cidr_subnet.rs

```diff
@@ -0,0 +1,342 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use super::Function;
+use crate::configure::context::Context;
+use crate::functions::{FunctionArgKind, FunctionCategory, FunctionMetadata};
+use crate::DscError;
+use ipnetwork::{IpNetwork, Ipv4Network, Ipv6Network};
+use rust_i18n::t;
+use serde_json::Value;
+use std::net::{Ipv4Addr, Ipv6Addr};
+use tracing::debug;
+
+#[derive(Debug, Default)]
+pub struct CidrSubnet {}
+
+impl Function for CidrSubnet {
+    fn get_metadata(&self) -> FunctionMetadata {
+        FunctionMetadata {
+            name: "cidrSubnet".to_string(),
+            description: t!("functions.cidrSubnet.description").to_string(),
+            category: vec![FunctionCategory::String],
```

Comment:
Looks like ARM put these under their own category `CIDR` so you should add a new category for this instead of under String